tests/unit/s2n_tls13_pq_handshake_test.c

```diff
@@ -284,95 +307,160 @@ int main()
+                .expected_curve = NULL,
+                .hrr_expected = false,
+                .len_prefix_expected = false,
+        },
+
+        /* Check that we're backwards and forwards compatible with different Hybrid PQ draft revisions*/
+        {
+                .client_policy = &kyber_test_policy_draft0,
+                .server_policy = &kyber_test_policy_draft5,
+                .expected_kem_group = expected_kyber_r3_group,
+                .expected_curve = NULL,
+                .hrr_expected = false,
+                .len_prefix_expected = true,
+        },
+        {
+                .client_policy = &kyber_test_policy_draft5,
+                .server_policy = &kyber_test_policy_draft0,
+                .expected_kem_group = expected_kyber_r3_group,
+                .expected_curve = NULL,
+                .hrr_expected = false,
+                .len_prefix_expected = false,
+        },
+        {
+                .client_policy = &security_policy_pq_tls_1_0_2021_05_24,
+                .server_policy = &security_policy_pq_tls_1_0_2023_01_24,
+                .expected_kem_group = expected_kyber_r3_group,
+                .expected_curve = NULL,
+                .hrr_expected = false,
+                .len_prefix_expected = true,
+        },
+        {
+                .client_policy = &security_policy_pq_tls_1_0_2023_01_24,
+                .server_policy = &security_policy_pq_tls_1_0_2021_05_24,
+                .expected_kem_group = expected_kyber_r3_group,
+                .expected_curve = NULL,
+                .hrr_expected = false,
+                .len_prefix_expected = false,
         },
 
         /* Server supports all KEM groups; client sends a PQ key share and an EC key
-             * share; server chooses to negotiate client's first choice PQ without HRR. */
+         * share; server chooses to negotiate client's first choice PQ without HRR. */
         {
                 .client_policy = &security_policy_pq_tls_1_0_2020_12,
                 .server_policy = &security_policy_pq_tls_1_0_2020_12,
                 .expected_kem_group = expected_kyber_r3_group,
                 .expected_curve = NULL,
                 .hrr_expected = false,
+                .len_prefix_expected = true,
+        },
+
+        /* Server supports only one KEM group and it is the client's first choice;
+         * client sends a PQ share and an EC share; server chooses to negotiate PQ
+         * without HRR. */
+        {
+                .client_policy = &security_policy_pq_tls_1_0_2020_12,
+                .server_policy = &kyber_test_policy_draft0,
+                .expected_kem_group = expected_kyber_r3_group,
+                .expected_curve = NULL,
+                .hrr_expected = false,
+                .len_prefix_expected = true,
         },
 
         /* Server supports only one KEM group and it is the client's first choice;
-             * client sends a PQ share and an EC share; server chooses to negotiate PQ
```

Comment:
Thanks for fixing these. clang-format really did a number on some of our comments :)